src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -706,11 +706,20 @@ int sizeOfEncodedHeaderAndPayloadAsciiBytes
 
             writer.WriteStartObject();
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
+            // TODO at next major version (8.0) use only Audiences as SecurityTokenDescriptor.Audience will be removed.
+            if (!tokenDescriptor.Audiences.IsNullOrEmpty())
             {
+                writer.WritePropertyName(JwtPayloadUtf8Bytes.Aud);
```

Comment:
Is it not possible to use JsonSerializerPrimitives.WriteStrings?

Reply:
when I did that it would end up as a collection of size one with the one string element being a json array of all the strings when deserialized. This way the strings each end up in their own index.

Reply:
I am not sure i am understanding.
The OpenIdConnectConfigurationSerializer writes out an array of strings.

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/b4560536c4abde119272bd173ea530e0211e26c3/src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Json/OpenIdConnectConfigurationSerializer.cs#L482

Reply:
figured it out and using a WriteStrings method now.